Move and Update Vue.js Docs (#5948)
PostHog/posthog.com: contents/docs/libraries/vue-js.md

**Vue 3.x:** 

```js
//./plugins/posthog.js
import posthog from "posthog-js";

export default {
  install(app) {
    app.config.globalProperties.$posthog = posthog.init(
      "<ph_project_api_key>",
      {
        api_host: "<ph_instance_address>",
      }
    );
  },
};
```

**Vue 2.x:** 

```js
//./plugins/posthog.js
import posthog from "posthog-js";

export default {
  install(Vue, options) {
    Vue.prototype.$posthog = posthog.init(
      "<ph_project_api_key>",
      {
        api_host: "<ph_instance_address>"
      }
    );
  }
};
```

Finally, activate your plugin in the file where you initialize your app. For Vue 3.x, it will likely be your `index.js` file. For Vue 2.x, it will be your `main.js` file.

**Vue 3.x:** 

```js
//main.js
import { createApp } from 'vue'
import App from './App.vue'
import posthogPlugin from "./plugins/posthog"; //import the plugin. 

const app = createApp(App);

app.use(posthogPlugin); //install the plugin
app.mount('#app')
```

**Vue 2.x:** 

```js
//main.js
import posthogPlugin from "./plugins/posthog"; // import the plugin

Vue.use(posthogPlugin); // install the plugin, before new Vue()
```

> You can now use PostHog throughout your Vue app using `this.$posthog`.  For example, 

Reviewer: ```suggestion
You can now use PostHog throughout your Vue app using `this.$posthog`. For example:
```

```js
//component.vue

<script>
export default {
  data() {
    return {
      foo: "bar!",
    };
  },
  watch: {
    // whenever question changes, this function will run
    foo(newFoo, oldFoo) {
      this.$posthog.capture("Foo Changed", {foo: newFoo});
    },
  },
  created() {
    console.log("Created", this.$posthog); //posthog accessible anywhere!
  },
};
</script>
```

### Method 2: Use provide / inject

_Note: For Vue 3.x users only_

With Vue 3.x, developers can use `provide()` and `inject()` to pipe global values into any component without prop drilling. And if you don’t know what prop drilling is, good for you.

> While this method is more declarative — as you need to inject PostHog into every component — it avoids “polluting” globals (such as in Method 1). Some engineers prefer this approach, while others to include PostHog in globals since it doesn’t need to be reactive and will be called throughout your application. 

Reviewer: ```suggestion
While this method is more declarative — as you need to inject PostHog into every component — it avoids “polluting” globals (like method 1 does). Some engineers prefer this approach, while others include PostHog in globals since it doesn’t need to be reactive and will be called throughout your application. 
```

#### Step 1: Initialize Vue

Prior to mounting the app, you must:
1. Import PostHog
2. Initialize it
3. Provide it to your app. 
   
This **must** be done *before* you mount my app. If I provide PostHog *after* mounting it, PostHog will not be predictably available. 

```js
//app.js

import posthog from "posthog-js";

const app = createApp(App);
posthog.init("<ph_project_api_key>", {
  api_host: "<ph_instance_addressT>",
});
app.provide("posthog", posthog);
```

#### Step 2: Inject into any Vue file

```js
//component.vue

export default {
  data() {
    return {
      greeting: "How are you!",
    };
  },
  inject: ["posthog"], //grab the injection from app-wide provider
  created() {
    console.log("Created", this.posthog); //posthog accessible!
  },
};
```
### Method 3: Use Vue.prototype

_Note: For Vue 2.x users only_ 

While Vue 3.x dramatically clamped down on global variables, in Vue 2.x, you can initialize PostHog by using `Object.defineProperty` and `Vue.prototype`. 

Anywhere, declare: 

```js
import posthog from "posthog-js";
Object.defineProperty(Vue.prototype, '$posthog', { value: posthog });
```

> Then, access PostHog. I can do this by calling `this.$posthog`. 

Reviewer: ```suggestion
Then, access PostHog by calling `this.$posthog`. 
```
